src/promptflow/promptflow/_sdk/_start_trace.py

```diff
@@ -0,0 +1,47 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import os
+import uuid
+
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+from promptflow._sdk._constants import TRACE_SESSION_ID_ENV_VAR
+
+
+def start_trace():
+    """Start a tracing session.
+
+    This will capture OpenAI and prompt flow related calls and persist traces;
+    it will also provide a UI url for user to visualize traces details.
+
+    Note that this function is still under preview, and may change at any time.
+    """
+    # detect PFS liveness
+    pfs_port = "55507"  # TODO: make this dynamic from PFS liveness probe
```

Comment:
need to refine logic here

Reply:
offline sync with Cle, I will take action here